cmd/snap-confine/ns-support.c

```diff
@@ -647,6 +676,16 @@ static void helper_capture_ns(struct sc_mount_ns *group, pid_t parent)
 	debug("capturing per-snap mount namespace");
 	sc_must_snprintf(src, sizeof src, "/proc/%d/ns/mnt", (int)parent);
 	sc_must_snprintf(dst, sizeof dst, "%s%s", group->name, SC_NS_MNT_FILE);
+	if (access(dst, F_OK) < 0 && errno == ENOENT) {
+		int fd =
+		    open(dst,
```

Comment:
out of curiosity, why wasn't this written as
```c
int fd = open(...);
if (fd < 0 && errno != EEXIST) { /* error out */ }
```
?

Reply:
No specific reason really. I guess I can simplify.